handlers/sf-move-subscriptions-api/src/main/scala/com/gu/sf/move/subscriptions/api/UpdateSupporterProductDataService.scala

```
@@ -0,0 +1,59 @@
+package com.gu.sf.move.subscriptions.api
+import cats.implicits.toTraverseOps
+import com.gu.effects.sqs.AwsSQSSend.{Payload, QueueName}
+import com.gu.effects.sqs.{AwsSQSSend, SqsSync}
+import com.gu.zuora.subscription.Subscription
+import com.typesafe.scalalogging.LazyLogging
+import io.circe.syntax.EncoderOps
+
+trait UpdateSupporterProductData {
+  def update(subscription: Subscription, identityId: String): Either[String, Unit]
+}
+
+class UpdateSupporterProductDataService(queueName: QueueName) extends UpdateSupporterProductData with LazyLogging {
+
+  def combineErrorsOrUnit(list: List[Either[String, Unit]]): Either[String, Unit] = {
+    list.foldLeft[Either[String, Unit]](Right(())) { (acc, either) =>
+      acc match {
+        case Left(errors) =>
+          either match {
+            case Left(error) => Left(s"$errors, $error")
+            case Right(_) => acc
+          }
+        case Right(_) =>
+          either match {
+            case Left(error) => Left(s"$error")
+            case Right(_) => acc
+          }
+      }
+    }
+  }
+  override def update(
+      subscription: Subscription,
+      identityId: String,
+  ): Either[String, Unit] = {
+    val sqsSync = SqsSync.send(SqsSync.buildClient)(queueName) _
+    logger.info(s"Attempting to write to SQS ${queueName.value}, events: ${subscription.ratePlans}")
+    val events = subscription.ratePlans
+      .map(ratePlan =>
+        SupporterRatePlanItem(
+          subscription.subscriptionNumber,
+          identityId,
+          ratePlan.productRatePlanId,
+          ratePlan.ratePlanName,
+          subscription.termEndDate,
+          subscription.contractEffectiveDate,
+        ),
+      )
+      .map(item => Payload(item.asJson.noSpaces))
+      .map(payload => sqsSync(payload).toEither.left.map(_.getMessage))
+
+    combineErrorsOrUnit(events)
```

not sure if it's common to have more than one here with only one failing - if there are a few to send it won't be clear which one the error is from. consider having e.g.  `0: OK, 1: IllegalArgumentException, 2: CredentialException, 3: OK, 4: OK` so we can debug?

---

on the flip side, if it's usually going to be a test with no credentials, having a page of errors to tell you that, fail fast could be better (given I don't think we're running in parallel)